tools/pylib/_boutcore_build/boutcore.pyx.in

```diff
@@ -813,7 +813,7 @@ do
             fu.cobj=f$n(<double?>float(self),(<Field2D?>other).cobj)
         else:
             print("$n",type(self),type(other),isinstance(self,numbers.Number))
-            return NotImplemented
+            raise NotImplemented("Unexpected other - not supported (yet?).")
```

Comment:
I might be wrong, but I thought that returning NotImplemented would be the right thing here (rather than raising NotImplemented(Error) )
Related SO answer:
https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror

Reply:
In python3 NotImplemented is also valid for `__radd__` and the likes.
See https://docs.python.org/3/library/constants.html#NotImplemented

Reply:
Cool. My mistake, I did not know about this usage and got confused by the raises a few lines below. Will go revert now...
sorry!

Reply:
Actually, if `return NotImplemented` is the right thing, is there a reason why it's `raise` instead of `return` just below?
https://github.com/boutproject/BOUT-dev/blob/655ad714572804a756af0810c03298a7b4c746e2/tools/pylib/_boutcore_build/boutcore.pyx.in#L431-L450

Reply:
No, not quite sure.
I think I had trouble as it did not quite work as expected, but that might have been because I was using python 2, rather than 3 at that time ...
It might well be a bug ... and a reason why more tests would be great, so that it can be fixed, without the fear of breaking stuff ... sorry for not writing them (yet?)

Reply:
I found this link too https://bugs.python.org/issue32464 which says it should be either `return NotImplemented` or `raise NotImplementedError`.